builder/azure/arm/azure_client.go

```diff
@@ -230,13 +230,12 @@ func NewAzureClient(ctx context.Context, isVHDBuild bool, cloud *environments.En
 	azureClient.GalleryImagesClient = *galleryImagesClient
 
 	// We only need the Blob Client to delete the OS VHD during VHD builds
-	if isVHDBuild {
+	if storageAccountName != "" {
 		storageAccountAuthorizer, err := commonclient.BuildStorageAuthorizer(ctx, authOptions, *cloud)
 		if err != nil {
 			return nil, err
 		}
-
-		blobClient, err := giovanniBlobStorageSDK.NewWithBaseUri(cloud.Storage.Name())
```

Comment:
This actually was just passing in "AzureStorage" as a string, however the client never invoked this so we never ran into the issue since Delete blobs endpoint previously accepted storage account, now we had to give it this URL.

See https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview#standard-endpoints for more info